src/apatheia/jobs.nim

```diff
@@ -66,25 +76,79 @@ proc newJobQueue*[T](maxItems: int = 0, taskpool: Taskpool = Taskpool.new()): Jo
   result = JobQueue[T](queue: newSignalQueue[(uint, T)](maxItems), taskpool: taskpool, running: true)
   asyncSpawn(processJobs(result))
 
+template checkJobArgs*[T](exp: seq[T], fut: untyped): OpenArrayHolder[T] =
+  # static:
+  #   echo "checkJobArgs::SEQ: ", $typeof(exp)
+  let rval = SeqHolder[T](data: exp)
+  let expPtr = OpenArrayHolder[T](data: cast[ptr UncheckedArray[T]](unsafeAddr(rval.data[0])), size: rval.data.len())
+  fut.addCallback proc(data: pointer) =
+    ## just to keep the rval GC object alive for the duration of the job
+    discard rval.data.len()
+    echo "FREE RVaL: ", rval.data.len()
```

Comment:
@tbekas @gmega so this setup is to capture a `seq[byte]` in a controlled manner. 

If we put the `seq[byte]` in a ref object and keep it alive in future, then passing a pointer to it using `openArray` will let us avoid needing to copy the data. 

However, I'm not sure what to do in the case of cancellations. 

There's `future.cancelCallback`, but if the task is running we can't just free the future and its callbacks in case the GC would free the seq memory while the task is using it.